Plugins/Covfie/src/FieldConversion.cpp

```diff
@@ -0,0 +1,181 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/Covfie/FieldConversion.hpp"
+
+namespace Acts::CovfiePlugin {
+
+/// @brief Get the value of the interpolated field at a specific position (unchecked).
+/// @tparam cache_t
+/// @param magneticField
+/// @param cache
+/// @param position
+/// @return
+template <typename cache_t>
+auto fieldLookup(const Acts::InterpolatedMagneticField& magneticField,
+                 [[maybe_unused]] cache_t& cache,
+                 const Acts::Vector3& position) {
+  return magneticField.getFieldUnchecked(position);
+}
+
+/// @brief Get the value of the field at a specific position of a general magnetic field.
+/// @tparam cache_t
+/// @param magneticField
+/// @param cache
+/// @param position
+/// @return
+template <typename cache_t>
+auto fieldLookup(const Acts::MagneticFieldProvider& magneticField,
+                 cache_t& cache, const Acts::Vector3& position) {
+  auto lookupResult = magneticField.getField(position, cache);
+  if (!lookupResult.ok()) {
+    throw std::runtime_error{"Field lookup failure"};
+  }
+  return *lookupResult;
+}
```

Comment:
What are these used for? There's already a type-erasure mechanism in place for the magnetic field caching, so templating should not be necessary here. Also, it seems like it's simply a wrapper around the underlying function call?